jdbc/src/ragtime/jdbc.clj

```diff
@@ -35,32 +35,45 @@
                        (str (:table_schem metadata) "." (:table_name metadata))
                        (:table_name metadata))))
 
-(defn- table-exists? [db-spec ^String table-name]
+(defn- table-exists-via-metadata-scan? [db-spec ^String table-name]
   (some (partial metadata-matches-table? table-name)
         (get-table-metadata db-spec)))
 
-(defn- ensure-migrations-table-exists [db-spec migrations-table]
-  (when-not (table-exists? db-spec migrations-table)
+(defn- table-exists-via-provided-sql? [db-spec
+                                       ^String migrations-table-exists-sql]
+  (pos? (count (sql/query db-spec [migrations-table-exists-sql]))))
```

Comment:
I don't think the `^String` tag is doing anything here, and the `migrations-table-exists-sql` can be shortened without loss of intent as the function name is sufficient to provide context.

So:

```clojure
(defn- table-exists-via-provided-sql? [db-spec sql]
  (pos? (count (sql/query db-spec [sql]))))
```